cli/src/utils/abortController.js

```diff
@@ -0,0 +1,10 @@
+export function createAbortSignal() {
+  const abortController = new AbortController();
+  const { signal: signalAbort } = abortController;
+  const handleAbort = () => {
+    abortController.abort();
+    process.off('SIGINT', handleAbort);
+  };
+  process.on('SIGINT', handleAbort);
+  return signalAbort;
+}
```

Comment:
SIGINT is trapped until SIGINT is emitted, there is no way to remove the SIGINT listener when the abort signal is no longer needed

I would implement this way

```suggestion
export function createSigintAbortSignal() {
  const abortController = new AbortController();
  function clearListener() {
    process.off('SIGINT', handleAbort);
  }
  function handleAbort() {
    abortController.abort();
    clearListener();
  }
  process.on('SIGINT', handleAbort);
  return {
    /**
     * AbortSignal trigged by SIGINT
     */
    signal: abortController.signal,
    /**
     * clear the SIGINT listener
     */
    clear: clearListener,
  };
}
```

in client code

```js
const { signal, clear } = createSigintAbortSignal();
doStruff({abortSignal: signal});
// wait doStuff has finished to clear the listener
clear();
```